File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let toRna (dna: string): string option = 
    let dna2rna x =
        match x with
        | 'G' -> 'C'
        | 'C' -> 'G'
        | 'T' -> 'A'
        | 'A' -> 'U'
        | _ -> '?'
    let rna = dna |> Seq.map dna2rna |> System.String.Concat
    match rna with
    | _ when rna.IndexOf("?") >= 0 -> None
    | _ -> Some rna

File: go/rna-transcription/rna_transcription.go
package strand

func ToRNA(dna string) string {
	result := ""
	for _, d := range dna {
		switch d {
		case 'G':
			result += "C"
		case 'C':
			result += "G"
		case 'T':
			result += "A"
		case 'A':
			result += "U"
		}
	}
	return result
}
